x-pack/test/functional/apps/index_management/data_streams_tab/data_streams_tab.ts

```diff
@@ -224,5 +225,171 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
+        // Navigate to the templates tab
+        await pageObjects.indexManagement.changeTabs('templatesTab');
+        await pageObjects.header.waitUntilLoadingHasFinished();
+        // Edit template
+        await pageObjects.indexManagement.clickIndexTemplateNameLink(
+          `index_template_${TEST_DS_NAME}`
+        );
+        await testSubjects.click('manageTemplateButton');
+        await testSubjects.click('editIndexTemplateButton');
+
+        // Verify index mode is origin
+        expect(await testSubjects.getVisibleText('indexModeField')).to.be(indexModeName);
+      };
+
+      const changeIndexMode = async (indexModeSelector: string) => {
+        // Modify index mode
+        await testSubjects.click('indexModeField');
+        await testSubjects.click(indexModeSelector);
+      };
+
+      const verifyModeHasBeenChanged = async (indexModeName: string) => {
+        expect(await testSubjects.getVisibleText('indexModeValue')).to.be(indexModeName);
+
+        // Click update template
+        await pageObjects.indexManagement.clickNextButton();
+
+        // Verify index mode and close detail tab
+        expect(await testSubjects.getVisibleText('indexModeValue')).to.be(indexModeName);
+        await testSubjects.click('closeDetailsButton');
+
+        // Navigate to the data streams tab
+        await pageObjects.indexManagement.changeTabs('data_streamsTab');
+        await pageObjects.header.waitUntilLoadingHasFinished();
+
+        // Open data stream
+        await pageObjects.indexManagement.clickDataStreamNameLink(TEST_DS_NAME);
+        // Check that index mode detail exists and its label is destination index mode
+        expect(await testSubjects.exists('indexModeDetail')).to.be(true);
+        expect(await testSubjects.getVisibleText('indexModeDetail')).to.be(indexModeName);
+        // Close flyout
+        await testSubjects.click('closeDetailsButton');
+      };
+
+      afterEach(async () => {
+        await log.debug('Cleaning up created data stream');
+
+        try {
+          await es.indices.deleteDataStream({ name: TEST_DS_NAME });
+          await es.indices.deleteIndexTemplate({
+            name: `index_template_${TEST_DS_NAME}`,
+          });
+        } catch (e) {
+          log.debug('Error deleting test data stream');
+          throw e;
+        }
+      });
+
+      it('allows to upgrade data stream from standard to logsdb index mode', async () => {
+        await setIndexModeTemplate({
+          mode: 'standard',
+        });
+        await verifyIndexModeIsOrigin('Standard');
+
+        await changeIndexMode('index_mode_logsdb');
+        // Navigate to the last step of the wizard
+        await testSubjects.click('nextButton');
+        await testSubjects.click('nextButton');
+        await testSubjects.click('nextButton');
+        await testSubjects.click('nextButton');
+        await testSubjects.click('nextButton');
+
+        await verifyModeHasBeenChanged('LogsDB');
```

Comment:
Instead of using strings for the index type, it would be nice to define the strings in one place, perhaps using an enum.